provider/datagen/src/transform/cldr/source.rs

```diff
@@ -222,3 +227,55 @@ impl<'a> CldrDirLang<'a> {
         }
     }
 }
+
+#[cfg(feature = "icu_transliteration")]
+pub(crate) struct CldrDirTransform<'a>(&'a CldrCache, String);
+
+#[cfg(feature = "icu_transliteration")]
+impl<'a> CldrDirTransform<'a> {
+    pub fn read_and_parse_metadata(
+        &self,
+        transform: &str,
+    ) -> Result<&'a crate::transform::cldr::cldr_serde::transforms::Resource, DataError> {
+        let dir_suffix = self.0.dir_suffix()?;
+        let path = format!("{}-{dir_suffix}/main/{transform}/metadata.json", self.1);
+        if self.0.serde_cache.file_exists(&path)? {
+            self.0.serde_cache.read_and_parse_json(&path)
+        } else {
+            Err(DataErrorKind::Io(std::io::ErrorKind::NotFound)
+                .into_error()
+                .with_display_context(&path))
+        }
```

Comment:
You should be able to construct a `CldrDirLang` on the fly to call its `read_and_parse`

Reply:
I assume you meant `CldrDirNoLang`, because there is no language identifier around. Doing this now:

```rust
let cldr_dir = CldrDirNoLang(self.0, format!("{}-{dir_suffix}/main/{transform}", self.1));
cldr_dir.read_and_parse("metadata.json")
```